trackhubs/tests/test_parser.py

```diff
@@ -0,0 +1,51 @@
+"""
+.. See the NOTICE file distributed with this work for additional information
+   regarding copyright ownership.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+
+import logging
+import pytest
+from trackhubs.parser import parse_file_from_url
+
+# disable logging when running tests
+logging.disable(logging.CRITICAL)
+
+
+def test_fetch_hub_success():
+    hub_url = 'https://data.broadinstitute.org/compbio1/PhyloCSFtracks/trackHub/hub.txt'
```

Comment:
This will bite you. Bear in mind it gets fetched for every single CI run. If Python will let you fake and HTTP server to host something on localhost, then that would be much safer.

I tried to beat this problem in the existing unit tests, but Perl couldn't trivially fake the relevant service without horrors.

Reply:
My bad, the function name here is misleading, I changed it to `test_parse_hub_success` because it's testing the parser, not the HTTP Response.

The tricky part in this test is that `parse_file_from_url(hub_url, ...)` function accepts a hub URL which will be read/opened as an  `http.client.HTTPResponse` object; I couldn't find a way to use a local text file and let `parse_file_from_url()` read it as an URL (all the use cases I found online are about mocking an API response, which isn't the purpose of this test).
```
>>> import urllib.request
>>> url = "http://urgi.versailles.inra.fr/repetdb/repetdb_trackhubs/repetdb_Melampsora_larici-populina_98AG31_v1.0/hub.txt"
>>> urllib.request.urlopen(url)
<http.client.HTTPResponse object at 0x106d36940>
```
What I did for now, is I used the text file hosted in GitHub's THR project